openedx/core/djangoapps/edx_global_analytics/tasks.py

```diff
@@ -42,22 +112,12 @@ def count_data():
     else:
         latitude, longitude = ip_data_json['latitude'], ip_data_json['longitude']
 
-    # Get count of active students (if logged in during last 30 days).
-    # Built-in edX`s accounts have None in `last_login` field.
-    # 30th day <= Current last login date <= Today
-    activity_border = (datetime.datetime.now() - datetime.timedelta(days=olga_settings.get("ACTIVITY_PERIOD")))
-    active_students_amount = UserProfile.objects.exclude(
-        Q(user__last_login=None) | Q(user__is_active=False)
-    ).filter(user__last_login__gt=activity_border).count()
-
-    # Aggregated students per country
-    students_per_country = UserProfile.objects.exclude(
-        Q(user__last_login=None) | Q(user__is_active=False)
-    ).filter(user__last_login__gt=activity_border).values('country'
-    ).annotate(count=Count('country'))
-
-    # Get courses amount within current platform.
-    courses_amount = len(modulestore().get_courses())
+    # Get all information, that need to be calculated.
+    (active_students_amount_day,
+     active_students_amount_week,
+     active_students_amount_month,
+     courses_amount,
+     students_per_country) = calculate_instance_information()
```

Comment:
Why not to call `calculate_instance_information` in SEND method and avoid transferring bunch of parameters?